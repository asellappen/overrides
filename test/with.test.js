import withOverrides from "../src/with";

import React from "react";
import renderer from "react-test-renderer";

const Bar = props => <div {...props}>Bar</div>;
const Baz = props => <div {...props}>Baz</div>;

const defaultComponents = { Bar };
const Foo = withOverrides(defaultComponents)(({ Bar, ...props }) => (
  <Bar {...props} />
));

describe("withOverrides", () => {
  it("should render without overrides", () => {
    const component = renderer.create(<Foo />);

    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should render with style overrides", () => {
    const component = renderer.create(
      <Foo overrides={{ Bar: { style: { color: "red" } } }} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should render with props overrides", () => {
    const component = renderer.create(
      <Foo overrides={{ Bar: { props: { className: "Bar" } } }} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should render with component overrides", () => {
    const component = renderer.create(
      <Foo overrides={{ Bar: { component: Baz } }} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
