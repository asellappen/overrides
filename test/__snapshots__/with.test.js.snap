// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withOverrides should render with component overrides 1`] = `
<div
  style={Object {}}
>
  Baz
</div>
`;

exports[`withOverrides should render with props overrides 1`] = `
<div
  className="Bar"
  style={Object {}}
>
  Bar
</div>
`;

exports[`withOverrides should render with style overrides 1`] = `
<div
  style={
    Object {
      "color": "red",
    }
  }
>
  Bar
</div>
`;

exports[`withOverrides should render without overrides 1`] = `
<div
  style={Object {}}
>
  Bar
</div>
`;
